packages/auth/ios/RNFBAuth/RNFBAuthModule.m

```diff
@@ -922,7 +927,10 @@ - (void)invalidate {
 - (FIRAuthCredential *)getCredentialForProvider:(NSString *)provider token:(NSString *)authToken secret:(NSString *)authTokenSecret {
   FIRAuthCredential *credential;
 
-  if ([provider compare:@"twitter.com" options:NSCaseInsensitiveSearch] == NSOrderedSame) {
+  // First check if we cached an authToken
+  if (credentials[authToken] != nil && ![credentials[authToken] isEqual:[NSNull null]]) {
```

Comment:
This looks to me like it alters getCredentialForProvider so that it has state-based + variable semantics. Wouldn't this affect existing code that didn't realize we might cache a credential now, and calls getCredentialForProvider after an apple error without calling invalidate in between?

Perhaps the error credential should be accessed some other way vs altering the semantics of this method?

Related: can you update the related documentation in the module's index.t.ds (where methods and parameters and auth values are documented) so developers will know how to use this?

Reply:
I think it's safe. The condition will only be true, when passing the updated credential that was received through `error.errorInfo.authCredential`.

Regarding documentation: I took a look and it seems the the error object is currently undocumented (e.g. that `error.userInfo` exists). I would suggest to add a `@throws` annotation to `linkWithCredential()` and add a type definition for the error object?

Reply:
Okay - that sounds reasonable and I'll trust you on that one - also sounds reasonable with regard to the documentation - I'd merge that - thanks @henrik-d